db.js

```diff
@@ -234,10 +274,10 @@ exports.init = function (sbot, config) {
           : null
         validate2.validateSingle(hmacKey, msgVal, latestMsgVal, (err, key) => {
           if (err) return cb(err)
+          updateState({key, value: msgVal})
```

Comment:
for this to be consistent there needs to be a change in addBatch as well. It only does updateState after log completes. I was thinking about this issue and I think this change is good. Would be nice with a test that shows the problem.

Reply:
if you want I have give it a go

Reply:
Oh, please do, I'm not entirely sure I understood, and a bit tired too